### DISCUSSION TITLE
CONNECT BY PRIOR equivalent in Trino

### DISCUSSION THREAD

**Discussion by vicks86** [trinodb/trino#14201] [General]
Hi,

Is there any equivalent function/query in Trino for "CONNECT BY PRIOR" / hierarchical query? 

Below is the Oracle sql query that I want to migrate into Starburst Trino Version: 370-e.1
  SELECT  a.CHILD_ID
    FROM HRCHY_MV a
    CONNECT BY PRIOR CHILD_ID = PARNT_ID
    START WITH a.PARNT_ID = 'EXT'

Thanks!

**Comment by findepi** [comment]
Would WITH RECURSIVE help?
see https://trino.io/docs/current/sql/select.html#with-recursive-clause

**Reply by vicks86** [reply]
Thanks!
I will check WITH RECURSIVE CTE.
